src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -26,12 +29,14 @@ internal class OAuth2Client
         private readonly Dictionary<string, string> _queryParameters = new Dictionary<string, string>();
         private readonly IHttpManager _httpManager;
         private readonly ITelemetryManager _telemetryManager;
+        private readonly IDeviceAuthManager _deviceAuthManager;
 
-        public OAuth2Client(ICoreLogger logger, IHttpManager httpManager, ITelemetryManager telemetryManager)
+        public OAuth2Client(ICoreLogger logger, IHttpManager httpManager, ITelemetryManager telemetryManager, IDeviceAuthManager deviceAuthManager)
         {
             _headers = new Dictionary<string, string>(MsalIdHelper.GetMsalIdParameters(logger));
             _httpManager = httpManager ?? throw new ArgumentNullException(nameof(httpManager));
             _telemetryManager = telemetryManager ?? throw new ArgumentNullException(nameof(telemetryManager));
+            _deviceAuthManager = deviceAuthManager;
```

Comment:
You already are using the "null object pattern", which is great. So I think you should use it here as well, i.e. if deviceAuthManager is null, use the NullDeviceAuthmanager. This will avoid you from having to continously check `if (deviceAuthManager != null) ...`

Reply:
Or make sure to always pass in a DeviceManager, that will also work. There is one attached to the service bundle...